### ISSUE TITLE
Consistent getter format

### ISSUE THREAD

**Issue by RX14** [NOVA-Team/NOVA-Core#175]
We should either use `getXXX()` or `XXX()` consistently throughout NOVA.

**Comment by kashike** [comment]
I'm a fan of `getXXX()` myself.

**Comment by Kubuxu** [comment]
As scala dev (same as @calclavia) we were promoting `xxx()` if there is not setter and `getXxx()/setXxx(value)` in case that setter exists.

**Comment by RX14** [comment]
I much prefer `getXXX()` and `setXXX()` because:

1. It's java convention: this is not a scala framework,
2. Groovy transforms getters/setters into properties

**Comment by calclavia** [comment]
What we could do:

Use set and get for everything in NOVA.
Create implicit wrappers in Scala.

**Comment by Kubuxu** [comment]
I don't know if that is possible only with implicit but some macro blackmagic might work.

**Comment by RX14** [comment]
:+1: for @calclavia's suggestion. Groovy can mixin to classes, I assume scala can too? Implicit classes?

**Comment by anti344** [comment]
```scala
implicit class SomethingWrapper(peer: Something){

  def xxx: X = peer.getX
  def xxx_=(x: X) = peer.setX(x)

  def yyy: Y = peer.getY
  def yyy_=(y: Y) = peer.setY(y)
}
```

**Comment by anti344** [comment]
```scala
import nova.Something
import novascala.SomethingWrapper

val something = new Something(xx, yy)
something.x = null
println(something.y)
```

**Comment by anti344** [comment]
There is also a BeanProperty annotation in Scala, but it's work in other way, adds Beans getters/setters to val/var 's

**Comment by Kubuxu** [comment]
@anti344 problem is that you would need wrapper per class. Macros should be able to solve it.

**Comment by anti344** [comment]
Not a macros then, full-blown scalac plugin, for converting field/field_= calls to getField/setField calls where possible. Also think of IDE support for this

**Comment by Kubuxu** [comment]
Macros are supported by all Scala IDEs and it is much cleaner solution as it is just code generation (it has nothing in common with C macros).

**Comment by RX14** [comment]
From the scala docs I don't see a way to add methods with macros...

**Comment by Kubuxu** [comment]
You could write implicit classes with macros.

**Comment by anti344** [comment]
But how would you.., oh i see on IRC, m-kay

**Comment by SoniEx2** [comment]
Use overloaded `XXX`. `XXX()` for getter, `XXX(YYY)` for setter.

**Comment by calclavia** [comment]
Alright, so I think we should settle for standard Java conventions. setX() getX()

**Comment by Caellian** [comment]
I agree with Calclavia, after all, NOVA is written in Java (primarily).
This was already talked about a lot and whole NOVA team agreed that
getXXX() and setXXX(YYY) will be used (correct me if I'm wrong).

**Comment by ExE-Boss** [comment]
I personally prefer `variable()` for getters of immutable values and `getVariable()`/`setVariable(value)` for getters/setters of mutable variables.
